.gitlab-ci.yml

```
@@ -1774,6 +1774,12 @@ deploy_windows_testing-a7:
     - .kitchen_datadog_agent_flavor
     - .kitchen_azure_location_north_central_us
 
+.kitchen_test_upgrade_iot_agent:
```

Nit: I'd name the jobs with an additional `7`:
```suggestion
.kitchen_test_upgrade7_iot_agent:
```
to match the other jobs' naming convention, and clearly show which configuration (upgrade7) is used in the test.